tests/test_api.py

```diff
@@ -229,66 +299,297 @@ def test_subsection_max_min(self, mock_graded_subsections):
+                )
+                rows = list(processor.get_iterator())
+        assert len(rows) == 2
 
         processor = api.GradeCSVProcessor(course_id=self.course_id, subsection=self.usage_key, subsection_grade_min=200)
         rows = list(processor.get_iterator())
         assert len(rows) != self.NUM_USERS + 1
 
+    @patch('lms.djangoapps.grades.api.graded_subsections_for_course_id')
+    def test_subsection_max_min_no_subsection_grade(self, mock_graded_subsections):
+        mock_graded_subsections.return_value = self._mock_graded_subsections()
+        processor = api.GradeCSVProcessor(course_id=self.course_id, subsection=self.usage_key, subsection_grade_max=101)
+        with patch('lms.djangoapps.grades.api.get_subsection_grades') as mock_subsection_grades:
+            with patch('lms.djangoapps.grades.api.CourseGradeFactory.read') as mock_course_grade:
+                mock_course_grade.return_value = Mock(percent=1)
+                mock_subsection_grades.side_effect = mock_subsection_grade(chain([None], repeat(make_mock_grade())))
+                rows = list(processor.get_iterator())
+        assert len(rows) == self.NUM_USERS
+
+    @patch('lms.djangoapps.grades.api.graded_subsections_for_course_id')
+    def test_no_subsection_grade(self, mock_graded_subsections):
+        mock_graded_subsections.return_value = self._mock_graded_subsections()
+        processor = api.GradeCSVProcessor(course_id=self.course_id, subsection=self.usage_key)
+        with patch('lms.djangoapps.grades.api.get_subsection_grades') as mock_subsection_grades:
+            with patch('lms.djangoapps.grades.api.CourseGradeFactory.read') as mock_course_grade:
+                mock_course_grade.return_value = Mock(percent=1)
+                mock_subsection_grades.side_effect = mock_subsection_grade(chain([None], repeat(make_mock_grade())))
+                rows = list(processor.get_iterator())
+        assert len(rows) == self.NUM_USERS +1
+        grade_column_index = rows[0].split(',').index('original_grade-homework')
+        row = rows[1].split(',')
+        assert row[grade_column_index] == ''
+
+        for i in (2, 3):
+            row = rows[i].split(',')
+            assert row[grade_column_index] == '1'
+
+
     @patch('lms.djangoapps.grades.api.CourseGradeFactory.read')
     def test_course_grade_filters(self, course_grade_factory_mock):
-        course_grade_factory_mock.return_value = Mock(percent=0.50)
+        course_grade_factory_mock.side_effect = cycle((Mock(percent=0.50), Mock(percent=0.70), Mock(percent=0.90)))
 
-        processor = api.GradeCSVProcessor(course_id=self.course_id, max_points=100, course_grade_min=60)
+        processor = api.GradeCSVProcessor(course_id=self.course_id, max_points=100, course_grade_min=50)
         rows = list(processor.get_iterator())
-        self.assertNotEqual(len(rows), self.NUM_USERS+1)
+        self.assertEqual(len(rows), self.NUM_USERS+1)
```

Comment:
this is the opposite of what it was before?

Reply:
It is, but I pretty much entirely changed the behavior of this test.

Reply:
Originally this first filter filtered out all results, and the second filtered none.
Now this filters none and the second filters all but one. Slightly confusing diff but in the end it's all good.